src/DynamoCore/Models/CustomNodeWorkspaceModel.cs

```diff
@@ -196,20 +196,23 @@ protected virtual void OnFunctionIdChanged(Guid oldId)
 
         public override bool SaveAs(string newPath, ProtoCore.RuntimeCore runtimeCore, bool isBackUp = false)
         {
-            var originalPath = FileName;
-
-            // A SaveAs to an existing function id prompts the creation of a new 
-            // custom node with a new function id
-            if (originalPath != newPath)
+            if (!isBackUp)
```

Comment:
Well, in terms of performance, it should be the same whether isBackup is true or not.

In terms of readability, it is just a few lines of code. 

But I will update the code.